editor/src/components/editor/actions/actions.tsx

```diff
@@ -4614,11 +4614,14 @@ export const UPDATE_FNS = {
       forkedFromProjectId: action.id,
     }
   },
-  SET_CURRENT_THEME: (action: SetCurrentTheme, editor: EditorModel): EditorModel => {
-    return {
-      ...editor,
-      theme: action.theme,
+  SET_CURRENT_THEME: (action: SetCurrentTheme, userState: UserState): UserState => {
+    const updatedUserConfiguration: UserConfiguration = {
+      shortcutConfig: userState.shortcutConfig,
```

Comment:
I think it's better if you spread the ...userState here, so if someone adds a third field, this code does not need updating here

Reply:
oh but I see, UserState !== UserConfiguration. eek!

Reply:
Agreed, and then the below return statement should also spread the result of this (so that you're not having to make the same change twice, even though it is minor right now) like so:
```ts
return {
  ...userState,
  ...updatedUserConfiguration,
}
```

Reply:
Urgh, yes, you are right, and we aren't storing the `UserConfiguration` separately. I think it still makes sense to spread the `updatedUserConfiguration` into the return value though